05-part_two.Rmd

```diff
@@ -36,37 +36,35 @@ get_max(c("a", "b", "c"))
+* Integration testing - tests the correct flow of data through the components.
+* System testing - tests the overall outputs make sense.
+* Acceptance testing - tests whether or not the work meets its objectives.
 
-Software testing can make a distinction between functional and non-functional testing too. In the context of an analytical model, a functional test would be one which tests the model produces what its intended to whereas a non-functional test would cover things such as its run-time and interface (i.e. things which affect the experience of using it, but not the numbers that come out).
+Software testing can make a distinction between functional and non-functional testing too. In the context of analysis, a functional test would be one which tests the analysis produces what its intended to whereas a non-functional test would cover things such as its run-time and interface (i.e. things which affect the experience of using it, but not the numbers that come out). For more information, see [here](https://www.guru99.com/levels-of-testing.html) or explore further with help from a search engine.
 
-For more information, see [here](https://www.guru99.com/levels-of-testing.html) or use your favourite search engine to explore the topic.
+As this is a chapter about testing code, we going to focus primarily on unit testing. The other testing aspects cover testing of wider areas of a software project than just the code, and other parts of the DfE QA framework cover how to handle these:
 
-As this is a chapter about testing code, we are only going to cover unit testing in more detail. The other testing aspects cover testing of wider areas of a software project than just the code, and other parts of the DfE QA framework cover how to handle these:
-
-* Integration testing tests correct data flow, transformations and interactions between different model components.
+* Integration testing tests correct data flow, transformations and interactions between different parts of the analysis.
 * System testing relates to verification tests on the overall outputs.
 * Acceptance testing relates to the validation pillar of the QA framework.
 
 ### Unit testing
 
-Unit testing seeks to test every distinct component within your code to enable to you isolate specific areas that may be introducing errors. If you have written you code using functions, it is sometimes easier to think of this as testing each specific function.
+Unit testing seeks to test every distinct code component to enable isolation of specific areas that may be introducing errors. For code made up of functions, it is sometimes easier to think of this as testing each specific function.
 
 One of the key principles of unit testing is that each unit (or function) should be tested independently of all the others and without dependency on any data sources that may change. The objective is to isolate the cause of any errors:
```

Comment:
```suggestion
One of the key principles of unit testing is that each unit (or function) should be tested independently of all the others and without dependency on any data sources that may change. The objective is to isolate the cause of any errors which means that unit tests should not rely on output from previous functions or rely on data from a live source (e.g SQL) as in both these instances it would not be possible to identify if the function being tested caused the error.
```